projects/plugins/jetpack/functions.opengraph.php

```diff
@@ -245,8 +245,24 @@ function jetpack_og_tags() {
 
 	foreach ( (array) $tags as $tag_property => $tag_content ) {
 		// to accommodate multiple images.
-		$tag_content = (array) $tag_content;
+
+		$is_multidimensional_array = false;
+		$stored_array              = null;
+		$tag_content               = (array) $tag_content;
+		// If the array is multidimensional, we need to unset and store the subarray and set a flag.
+		foreach ( $tag_content as $key => $string ) {
+			if ( is_array( $string ) ) {
+				$is_multidimensional_array = true;
+				$stored_array              = $string;
+				unset( $tag_content[ $key ] );
+			}
+		}
+		// Clean up the array.
 		$tag_content = array_unique( $tag_content );
```

Comment:
The `array_unique()` is not supposed to handle multidimensional arrays, and when that happens, we see the `Array to string conversion`. Since I was unable to reproduce the problem, and the code worked so far with multidimensional arrays (except the warning), I went with the solution to remove sub-arrays, clean up what is left with `array_unique()`, and then reattach the sub-array.
Another possible solution would be to flatten the multidimensional array, but it seemed to be a riskier option.

Reply:
This probably works to fix the one warning, but associated with these warnings from `array_unique()` you're probably seeing more from `esc_url()` or `esc_attr()` via lines 283 or 286 below too. In PHP 8 line 283 probably winds up fatalling (complaining about passing a non-string to `trim()`), while 286 will result in `content="Array"` in the OG tag.

Since the code here doesn't handle arrays at all, possibly the thing to do would be to filter them out entirely. Maybe like
```php
$tag_content = array_unique( array_filter( $tag_content, 'is_scalar' ) );
```
if we don't mind filtering out objects too, or you could do something like `function ( $v ) { return ! is_array( $v ); }` instead of `'is_scalar'` if you want to target arrays specifically.

P.S. Reproducing it probably involves using the 'jetpack_open_graph_base_tags' filter to inject something weird.

Reply:
Thank you for the suggestion! I believe the 'is_scalar' is good enough here. I don't see where we would expect to see and deal with an object.